Validate the CIDR address is IPv4 and the network mask boundary is correct

micahlee opened the pull request:
This PR updates our CIDR value validation to ensure:
* The CIDR is IPv4

* The given IP address does not include any bits to the right of the network mask (e.g. `10.0.0.1/24` instead of `10.0.0.0/24`)

Connected to #30

jvanderhoof reviewed (approved):
Looks good 😄 .

jonahx commented on lib/conjur/cidr.rb:
```diff
@@ -0,0 +1,29 @@
+  class CIDR
+    def initialize(cidr)
+      @cidr = cidr.to_s
+      @ipaddr = cidr.kind_of?(IPAddr) ? cidr : IPAddr.new(cidr)
```
I think rather than having flexible input here, I'd rather make the client be specific:

1. canonical `new` ctor for IPAddrs
2. a `from_s` ctor for strings, which converts to IP and then delegates to 1.

micahlee replied:
@jonahx, that actually doesn't work well in this case because string to `IPAddr` is lossy (where it strips the extra bits in an invalid CIDR input). We need to preserve the original string representation if we have one.

In this particular case, we don't actually need to accept input as an `IPAddr`, so I'll just remove that as YAGNI. If we find we need it later, we can the additional constructor then.

micahlee committed:
Validate the CIDR address and network mask boundary is correct

A valid CIDR subnet should not include any address bits to the right of the network
mask. This commit validates that and returns an error and suggested correction
when this is not the case.

micahlee committed:
Validate that the restricted_to value is IPv4

micahlee committed:
Update tested Ruby versions to current supported versions

Support for Ruby 2.3 and 2.4 has ended, so this
commit updates our test matrix to test 2.5-2.7.

micahlee committed:
Remove `releaser-v2` node label

This node label no longer exists and the standard `executor-v2` label is functionally
equivalent.
